packages/mobile/src/web3/saga.ts

```diff
@@ -31,47 +33,65 @@ import {
 } from 'src/web3/actions'
 import { web3 } from 'src/web3/contracts'
 import { refreshGasPrice } from 'src/web3/gas'
+import { web3ReadySelector } from 'src/web3/reducer'
 import { currentAccountSelector } from 'src/web3/selectors'
 import { Block } from 'web3/eth/types'
 
 const ETH_PRIVATE_KEY_LENGTH = 64
 const MNEMONIC_BIT_LENGTH = (ETH_PRIVATE_KEY_LENGTH * 8) / 2
 
 const TAG = 'web3/saga'
-// The timeout for web3 to complete syncing
-const CHECK_WEB3_SYNC_PROGRESS_TIMEOUT = 60000
 // The timeout for web3 to complete syncing and the latestBlock to be > 0
 const CHECK_SYNC_PROGRESS_TIMEOUT = 60000
 const BLOCK_CHAIN_CORRUPTION_ERROR = "Error: CONNECTION ERROR: Couldn't connect to node on IPC."
 
 let AssignAccountLock = false
 
+export function* waitForWeb3Ready(): any {
+  const connected = yield select(web3ReadySelector)
+  if (connected) {
+    return
+  }
+
+  while (true) {
+    const action = yield take(Actions.SET_IS_READY)
+    if (action.payload.isReady) {
+      return
+    }
+  }
+}
+
 // checks if web3 claims it is currently syncing or not
-function* checkWeb3SyncProgressClaim() {
-  Logger.debug(TAG, 'Checking sync progress claim')
+export function* checkWeb3SyncProgressClaim() {
+  // yield call(waitForGethConnectivity)
   while (true) {
+    yield delay(1000) // Doesn't need to update that fast
     try {
-      const syncProgress = yield web3.eth.isSyncing()
-      Logger.debug(TAG, 'Sync progress', syncProgress)
+      const syncProgress = yield web3.eth.isSyncing() // return false when it's still syncing and thus not ready
 
       if (typeof syncProgress === 'boolean' && !syncProgress) {
-        // For some weird reason, checkSyncProgressWorker is flaky and does not work for the long running
-        // sync tasks.
-        Logger.debug(TAG, 'sync complete')
-        yield put(setSyncProgress(100))
-        yield put(setIsReady(true))
+        if (!(yield select(web3ReadySelector))) {
```

Comment:
I don't understand why we check that here? 
When would we reach this point but have already set is ready to true?

Reply:
It's there because the loop is infinite for the lifecycle of the app.

Reply:
I don't think that's what we want though. If this saga is constantly running for the lifetime of the app, then it's wasting resources and can potentially interfere with other saga (we've seen that issue before)